### ISSUE TITLE
Swagger2; issue making array items nullable

### ISSUE THREAD

**Issue by erodewald** [victools/jsonschema-generator#195]
Hi, 

I'm evaluating adoption of the `Swagger2Module` (which seems pretty flexible!) and I noticed that I was unable to apply `nullable = true` to the items of an array.

### Configuration
Dependencies: `4.19.0`
Generator:
```java
Swagger2Module swagger2Module = new Swagger2Module();
SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(SchemaVersion.DRAFT_2019_09,
    OptionPreset.PLAIN_JSON)
    .with(Option.VALUES_FROM_CONSTANT_FIELDS,
        Option.MAP_VALUES_AS_ADDITIONAL_PROPERTIES,
        Option.FORBIDDEN_ADDITIONAL_PROPERTIES_BY_DEFAULT,
        Option.FLATTENED_OPTIONALS)
    .with(swagger2Module);
```
POJO:
```java
@ArraySchema(
    arraySchema = @Schema(description = "Description on root field"),
    schema = @Schema(
        nullable = true,
        description = "Description on individual items"))
private Set<String> features = Set.of();
```

#### Expectation
```json
"features" : {
    "description" : "Description on root field",
    "type" : "array",
    "items" : {
      "type" : ["string", "null"],
      "description" : "Description on individual items"
    }
}
```
#### Generated
```json
"features" : {
    "description" : "Description on root field",
    "type" : "array",
    "items" : {
      "type" : "string",
      "description" : "Description on individual items"
    }
}
```

Is it clear from my configuration why this is happening? Setting `@Schema(nullable = true)` seems to work for non-lists/arrays.

Thanks.

**Comment by CarstenWickner** [comment]
HI @erodewald,

This limitation is independent of the `Swagger2Module` actually.
The configuration for being "nullable" or not currently refers only to fields and methods, and not types in general such as the type of array items.

For backward compatibility reasons, array items are currently prohibited from being `nullable`:
https://github.com/victools/jsonschema-generator/blob/ff053453187ad5a99ac8ca7fe0e139de639f47b2/jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/impl/SchemaGenerationContextImpl.java#L572

Maybe a new `Option` could be added, to lift this limitation. One would just have to be more careful in the `isNullable` checks, i.e. always consider the `MemberScope.isFakeContainerItemScope`'s return value.

**Comment by erodewald** [comment]
Thanks, I was having trouble identifying this. Some superficial testing indicates this would work for _me_, but your warning gives me pause on implementing any `Option` flags—backwards compatibility is important.

**Comment by CarstenWickner** [comment]
As longs as the new `Option` is an opt-in, i.e. turned off by default, I see no harm in making that behavior change.
I've created a draft now in PR #196 – still without documentation though.

**Comment by CarstenWickner** [comment]
The change has just been released in version [4.20.0](https://github.com/victools/jsonschema-generator/releases/tag/v4.20.0).
The new behavior is only available when explicitly enabling the new `Option.NULLABLE_ARRAY_ITEMS_ALLOWED`.

**Comment by erodewald** [comment]
Fantastic! I go on one little weekend vacation and there it is! 🎉  Thank you @CarstenWickner, I will be sure to reach out if I can identify any issues.
